lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2769,9 +3204,14 @@ ActorIsolation ActorIsolationRequest::evaluate(
   // Check for instance members and initializers of actor types,
   // which are part of actor-isolated state.
   if (auto nominal = value->getDeclContext()->getSelfNominalTypeDecl()) {
-    if (nominal->isActor() &&
-        (value->isInstanceMember() || isa<ConstructorDecl>(value))) {
-      defaultIsolation = ActorIsolation::forActorInstance(nominal);
+    if (nominal->isActor()) {
+      if (value->isInstanceMember() || isa<ConstructorDecl>(value)) {
+        defaultIsolation = nominal->isDistributedActor() ?
+                           ActorIsolation::forDistributedActorInstance(nominal) :
```

Comment:
There's a distinction between "`distributed func`" and "`func` inside a `distributed actor`" that I'm not sure we're accounting for properly. The latter doesn't allow cross-actor calls at all.

Reply:
Sure